zerver/lib/outgoing_webhook.py

```diff
@@ -8,17 +8,60 @@
 import logging
 from six.moves import urllib
 from functools import reduce
+from requests import Response
 
 from django.utils.translation import ugettext as _
 
-from zerver.models import Realm, get_realm_by_email_domain, get_user_profile_by_id, get_client
+from zerver.models import Realm, UserProfile, get_realm_by_email_domain, get_user_profile_by_id, get_client
 from zerver.lib.actions import check_send_message
 from zerver.lib.queue import queue_json_publish
 from zerver.lib.validator import check_dict, check_string
 from zerver.decorator import JsonableError
 
 MAX_REQUEST_RETRIES = 3
 
+class OutgoingWebhookServiceInterface(object):
+
+    def __init__(self, base_url, token, user_profile, service_name):
+        # type: (Text, Text, UserProfile, Text) -> None
+        self.base_url = base_url  # type: Text
+        self.token = token  # type: Text
+        self.user_profile = user_profile  # type: Text
+        self.service_name = service_name  # type: Text
+
+    # Given an event that triggers an outgoing webhook operation, returns the REST
+    # operation that should be performed, together with the body of the request.
+    #
+    # The input format can vary depending on the type of webhook service.
+    # The return value should be a tuple (rest_operation, request_data), where:
+    # rest_operation is a dictionary containing the following keys: ...
```

Comment:
Sorry I missed it. Along with these comments, user can take a look at generic class to understand the working.